sockets/lib/commands.go

```diff
@@ -0,0 +1,110 @@
+/**
+ * Commands
+ * https://pokemonshowdown.com/
+ *
+ * Commands are an abstraction over IPC messages sent to and received from the
+ * parent process. Each message follows a specific syntax: a one character
+ * token, followed by any number of parametres separated by newlines. Commands
+ * give the multiplexer and IPC connection a simple way to determine which
+ * struct it's meant to be handled by, before enqueueing it to be distributed
+ * to workers to finally process their payload concurrently.
+ */
+
+package sockets
+
+import "strings"
+
+// IPC message types
+const (
+	SOCKET_CONNECT       byte = '*'
+	SOCKET_DISCONNECT    byte = '!'
+	SOCKET_RECEIVE       byte = '<'
+	SOCKET_SEND          byte = '>'
+	CHANNEL_ADD          byte = '+'
+	CHANNEL_REMOVE       byte = '-'
+	CHANNEL_BROADCAST    byte = '#'
+	SUBCHANNEL_MOVE      byte = '.'
+	SUBCHANNEL_BROADCAST byte = ':'
+)
+
+type Command struct {
+	token    byte      // Token designating the type of command.
+	paramstr string    // The command parametre list, unparsed.
+	count    int       // The number of parametres in the paramstr.
+	target   CommandIO // The target to process this command.
+}
+
+// The multiplexer and the IPC connection both implement this interface. Its
+// purpose is solely to allow the two structs to be used in Command.
+type CommandIO interface {
+	Process(Command) (err error) // Invokes one of its methods using the command's token and parametres.
```

Comment:
No need to name output variable name of interface, it's clear that `error` type is... an error.